### ISSUE TITLE
[BUG] Native Image compilation fails

### ISSUE THREAD

**Issue by barnavarga** [optimizely/java-sdk#539]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SDK Version

3.10.4

### Current Behavior

Optimizely Java SDK would have been used in our Spring applications that are compiled to native images but during the compilation it failed because there is no information about the parameter names of methods SDK's classes  and constructors in bytecode.

### Expected Behavior

Based on a preliminary investigation, adding `-parameters` flag to the compiler args can solve this issue. When the flag is added to the compilation of Optimizely Java SDK the compiled source is ready to use in native images.
`options.compilerArgs.addAll(['-parameters'])`

### Steps To Reproduce

1. Initiate a Spring Boot application with the support of native image. [https://start.spring.io/](https://start.spring.io/)
2. Define Optimizely's SDK as a dependency and set the required minimum settings to the project.
3. Try to build image: `mvn -Pnative spring-boot:build-image` or `gradle bootBuildImage`

### Java Version

Java 21

### Link

N/A

### Logs

```
Exception in thread "main" org.springframework.boot.context.properties.bind.MissingParametersCompilerArgumentException: Constructor binding in a native image requires compilation with -parameters but the following classes were compiled without it:
	com.optimizely.ab.optimizelydecision.OptimizelyDecideOption

	at org.springframework.boot.context.properties.bind.BindableRuntimeHintsRegistrar.registerHints(BindableRuntimeHintsRegistrar.java:100)
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanFactoryInitializationAotProcessor$ConfigurationPropertiesReflectionHintsContribution.applyTo(ConfigurationPropertiesBeanFactoryInitializationAotProcessor.java:74)
	at org.springframework.context.aot.BeanFactoryInitializationAotContributions.applyTo(BeanFactoryInitializationAotContributions.java:78)
	at org.springframework.context.aot.ApplicationContextAotGenerator.lambda$processAheadOfTime$0(ApplicationContextAotGenerator.java:58)
	at org.springframework.context.aot.ApplicationContextAotGenerator.withCglibClassHandler(ApplicationContextAotGenerator.java:67)
	at org.springframework.context.aot.ApplicationContextAotGenerator.processAheadOfTime(ApplicationContextAotGenerator.java:53)
	at org.springframework.context.aot.ContextAotProcessor.performAotProcessing(ContextAotProcessor.java:106)
	at org.springframework.context.aot.ContextAotProcessor.doProcess(ContextAotProcessor.java:84)
	at org.springframework.context.aot.ContextAotProcessor.doProcess(ContextAotProcessor.java:49)
	at org.springframework.context.aot.AbstractAotProcessor.process(AbstractAotProcessor.java:82)
	at org.springframework.boot.SpringApplicationAotProcessor.main(SpringApplicationAotProcessor.java:80)
```

### Severity

Blocking development

### Workaround/Solution

N/A → Managing a fork is not a future-proof solution.

**Comment by jaeopt** [comment]
@barnavarga I'm trying to reproduce the native-build error locally. I followed the steps you summarized, but I see BUILD completed successfully. These are steps I followed - 

1. Initiate a Spring Boot application with the support of native image. https://start.spring.io/ (I added "GraalVM Native Support" as dependency but keep all other options as default).
2. Define Optimizely's SDK as a dependency and set the required minimum settings to the project.
3. "gradle bootBuildImage". This build completes successfully.

I used our java-sdk 3.10.4 and Java21. Any step I missed?
